kubernetes/provider.go

```diff
@@ -258,6 +263,14 @@ func providerConfigure(d *schema.ResourceData, terraformVersion string) (interfa
 		cfg.ExecProvider = exec
 	}
 
+	// Attempt to load in-cluster config if no other configuration was provided
+	if !hasConfig {
+		cfg, err = restclient.InClusterConfig()
```

Comment:
Hmm, [`DeferredLoadingClientConfig.ClientConfig()`](https://github.com/kubernetes/client-go/blob/v12.0.0/tools/clientcmd/merged_client_builder.go#L97) which is invoked from [tryLoadingConfigFile()](https://github.com/terraform-providers/terraform-provider-kubernetes/pull/686/files#diff-da3a5957d1adf1d97d4dec9f43b36ec1R331)
is already supposed to handle the in-cluster configuration case when no other configuration was provided: https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/merged_client_builder.go#L112-L117

Reply:
I've run into that too. That whole package is a bit of quagmire (k8s.io/client-go/tools/clientcmd). I'm still not sure if using those top-level helpers is worth it or not. Might also be the reason the config loading logic in this provider is quite custom too.